### ISSUE TITLE
Feature Request: Style/EachForSimpleLoop

### ISSUE THREAD

**Issue by deivid-rodriguez** [bbatsov/rubocop#3018]
I'd like RuboCop to detect and autocorrect things like:

```ruby
(1..8).each do |_n|
   puts "Do something that does not use the block parameter..."
   # ...
end
```

The preferred style would be

```ruby
8.times do
  puts "Do something that does not use the block parameter..."
  # ...
end 
```

I guess it could be generalized to

```ruby
collection.each do |_item|
  puts "Do something that does not use `_item` or mutate `collection`"
  # ...
end
```

Autocorrected the same way.

Does this sound like a good idea?

**Comment by Drenmi** [comment]
Hi @deivid-rodriguez!

There's one called `Lint/UnusedBlockArgument`. Is this something like what you're looking for? It doesn't have auto-correct, though.

https://github.com/bbatsov/rubocop/blob/master/lib/rubocop/cop/lint/unused_block_argument.rb

**Comment by deivid-rodriguez** [comment]
Hi @Drenmi! No, it's not that. The code I'm talking about has already corrected by the cop you mention.

I'm looking for a smarter cop that detects that you're using things like `(1..10).each { |n| ... }` when you could use `10.times { ... }`. It could reuse the `Lint/UnusedBlockArgument` though since a necessary condition for this replacement to make sense is that the block argument is unused.

**Comment by alexdowad** [comment]
> a necessary condition for this replacement to make sense is that the block argument is unused

Actually, it's not a necessary condition, because `times` also yields numbers from 0 up to N. But many people might find that less readable.

**Comment by deivid-rodriguez** [comment]
Right, I find the `times` version more readable only when the iteration index is unused.
